packages/components/src/font-size-picker/types.ts

```diff
@@ -81,18 +76,25 @@ export type FontSize = {
 	slug: string;
 };
 
-export type FontSizeOption = Omit< FontSize, 'size' > &
-	Partial< Pick< FontSize, 'size' > >;
-
-export type FontSizeSelectOption = Pick< FontSizeOption, 'size' > & {
-	key: string;
-	name?: string;
-	__experimentalHint: ReactNode;
+export type FontSizePickerSelectProps = {
+	fontSizes: FontSize[];
+	value: FontSizePickerProps[ 'value' ];
+	size: FontSizePickerProps[ 'size' ];
+	onChange: ( value: number | string | undefined ) => void;
+	onSelectCustom: () => void;
 };
 
-export type FontSizeToggleGroupOption = {
+export type FontSizePickerSelectOption = {
 	key: string;
-	value: number | string;
-	label: string;
 	name: string;
+	value?: FontSize[ 'size' ];
```

Comment:
Is there any particular reason why this prop name was changed from `size` to `value` ? It doesn't look like this affects `CustomSelectControl`, but was curious to know the reason.

Reply:
Two reasons.

These things are called `value` once given to `onChange`:

https://github.com/WordPress/gutenberg/blob/bbf17a3f5c90f488bbb9b6b34146d5b3cadca859/packages/components/src/font-size-picker/types.ts#L25

It makes it consistent with how the `ToggleGroupControl` works where each `ToggleGroupControlOption` has a `value` prop:

https://github.com/WordPress/gutenberg/blob/bbf17a3f5c90f488bbb9b6b34146d5b3cadca859/packages/components/src/font-size-picker/toggle-group.tsx#L48

Hopefully soon we can make `CustomSelectControl` work more like `ToggleGroupControl` in that we have a `CustomSelectControlOption` component. It would make this component much nicer to read imo 😀

Reply:
> Hopefully soon we can make CustomSelectControl work more like ToggleGroupControl in that we have a CustomSelectControlOption component. It would make FontSizePickerSelect much more comprehensible imo 😀

I hope I'll be able to resume the work on `CustomSelectControl` soon enough, although I'm not sure if we're going to be able to have `option`-like children, given the data-processing (ie. filtering) needs of the component (see [this comment](https://github.com/WordPress/gutenberg/pull/41952#issuecomment-1171140086))